apps/storefront/src/pages/QuoteDraft/index.test.tsx

```diff
@@ -1275,6 +1284,336 @@ describe('when the user is a B2B customer', () => {
+          global: buildGlobalStateWith({
+            blockPendingQuoteNonPurchasableOOS: { isEnableProduct: false },
+            featureFlags: {
+              'B2B-3318.move_stock_and_backorder_validation_to_backend': true,
+            },
+            quoteSubmissionResponse: {
+              value: '0',
+            },
+          }),
+        },
+      });
+
+      server.use(
+        graphql.mutation('CreateQuote', () =>
+          HttpResponse.json(
+            {
+              errors: [
+                {
+                  path: ['quoteCreate'],
+                  extensions: {
+                    productValidationErrors: [
+                      {
+                        itemId: '1abc070d-8c4c-4f0d-9d8a-162843c10333',
+                        productId: 112,
+                        variantId: 77,
+                        responseType: 'ERROR',
+                        code: 'OOS',
+                        productName: 'out of stock prod',
+                      },
+                    ],
+                  },
+                },
+              ],
+            },
+            { status: 400 },
+          ),
+        ),
+      );
+
+      await waitForElementToBeRemoved(() => screen.queryByText(/loading/i));
+      const submitButton = await screen.findByRole('button', { name: /Submit/i });
+      expect(submitButton).toBeInTheDocument();
+
+      await userEvent.click(submitButton);
+
+      navigation.mockClear();
+
+      expect(window.b2b.callbacks.dispatchEvent).toHaveBeenCalledWith('on-quote-create', {
+        channelId: 1,
+        companyId: 0,
+        storeHash: 'store-hash',
+        userEmail: customerEmail,
+        subtotal: '1000.00',
+        contactInfo: quoteInfo.draftQuoteInfo.contactInfo,
+        taxTotal: '500.00',
+        productList: [
+          {
+            basePrice: '10.00',
+            offeredPrice: '10.00',
+            discount: '0.00',
+            imageUrl: 'url',
+            itemId: expect.stringMatching(/\w{8}-\w{4}-\w{4}-\w{4}-\w{12}/),
+            options: [],
+            productId: 4451490883947128,
+            productName: 'Unbranded Rubber Cheese',
+            quantity: 100,
+            sku: 'test',
+            variantId: undefined,
+          },
+        ],
+        quoteTitle: '',
+        recipients: [],
+        discount: '0.00',
+        extraFields: [],
+        fileList: [],
+        legalTerms: '',
+        shippingAddress: noAddress,
+        billingAddress: noAddress,
+        referenceNumber: '123',
+        currency: {
+          currencyCode: 'USD',
+          currencyExchangeRate: '1.0000000000',
+          decimalPlaces: 2,
+          decimalToken: '.',
+          location: 'left',
+          thousandsToken: ',',
+          token: '$',
+        },
+        message: 'meow',
+        grandTotal: '1000.00',
+        totalAmount: '1500.00',
+      });
+
+      expect(getVariantInfoOOSAndPurchase).not.toHaveBeenCalled();
+
+      setTimeout(() => {}, 1000);
```

Comment:
do we really need this timeout? I'd expect the findByText to wait for the snackbar 🤔

Reply:
mmh no, sorry, I forgot to delete it